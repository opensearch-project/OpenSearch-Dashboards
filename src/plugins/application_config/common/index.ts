export const PLUGIN_ID = 'applicationConfig';
export const PLUGIN_NAME = 'applicationConfig';
