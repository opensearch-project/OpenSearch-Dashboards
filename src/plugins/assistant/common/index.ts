/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'assistant';
export const PLUGIN_NAME = 'assistant';

export * from './types';
