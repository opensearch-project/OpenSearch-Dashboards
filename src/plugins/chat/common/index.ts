/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'chat';
export const PLUGIN_NAME = 'chat';
export const CHAT_DEFAULT_AG_UI_URL = 'http://localhost:3000';
