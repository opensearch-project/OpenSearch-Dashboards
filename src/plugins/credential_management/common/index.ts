/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'credentialManagement';
export const PLUGIN_NAME = 'credentialManagement';
