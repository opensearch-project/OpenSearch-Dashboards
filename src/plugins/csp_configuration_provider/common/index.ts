/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_ID = 'cspConfigurationProvider';
export const PLUGIN_NAME = 'CspConfigurationProvider';
