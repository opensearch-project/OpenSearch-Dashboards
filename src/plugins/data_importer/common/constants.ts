/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const PLUGIN_NAME_AS_TITLE = 'Data Importer';
export const PLUGIN_ID = 'dataImporter';
export const PLUGIN_NAME = 'dataImporter';

export const CSV_FILE_TYPE = 'csv';
export const JSON_FILE_TYPE = 'json';
export const NDJSON_FILE_TYPE = 'ndjson';

export const DEFAULT_SUPPORTED_FILE_TYPES_LIST = [CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE];

export const CSV_SUPPORTED_DELIMITERS = [',', ';', '\t', '|'];

export enum DYNAMIC_MAPPING_TYPES {
  NULL = 'null',
  BOOLEAN = 'boolean',
  FLOAT = 'float',
  DOUBLE = 'double',
  INTEGER = 'integer',
  OBJECT = 'object',
  ARRAY = 'array',
  TEXT = 'text',
  KEYWORD = 'keyword',
  DATE = 'date',
}
