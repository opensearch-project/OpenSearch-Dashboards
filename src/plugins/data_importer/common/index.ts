export const PLUGIN_ID = 'dataImporter';
export const PLUGIN_NAME = 'Data Importer';
