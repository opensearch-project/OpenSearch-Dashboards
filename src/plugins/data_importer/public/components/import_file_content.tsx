/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiFilePicker } from '@elastic/eui';
import React from 'react';
import uuid from 'uuid';

export interface ImportFileContentBodyProps {
  enabledFileTypes: string[];
  onFileUpdate: (file?: File) => void;
}

export const ImportFileContentBody = ({
  enabledFileTypes,
  onFileUpdate,
}: ImportFileContentBodyProps) => {
  const acceptedFileExtensions = enabledFileTypes.map((fileType) => `.${fileType}`).join(', ');

  const onFileChange = (files: FileList | null) => {
    if (files && files.length > 0) {
      onFileUpdate(files[0]);
    } else {
      onFileUpdate(undefined);
    }
  };

  return (
    <EuiFilePicker
      id={uuid.v4()}
      fullWidth={true}
      display={'large'}
      accept={acceptedFileExtensions}
      onChange={onFileChange}
      initialPromptText={`Select or drag and drop a ${acceptedFileExtensions} file`}
    />
  );
};
