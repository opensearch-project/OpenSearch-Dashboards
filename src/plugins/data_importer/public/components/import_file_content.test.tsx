/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ImportFileContentBody } from './import_file_content';
import { DEFAULT_SUPPORTED_FILE_TYPES_LIST } from '../../common';

jest.mock('uuid', () => ({
  v4: jest.fn(() => 'mocked-id'),
}));

describe('ImportFileContent', () => {
  it('should render', () => {
    const { container } = render(
      <ImportFileContentBody
        enabledFileTypes={DEFAULT_SUPPORTED_FILE_TYPES_LIST}
        onFileUpdate={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
