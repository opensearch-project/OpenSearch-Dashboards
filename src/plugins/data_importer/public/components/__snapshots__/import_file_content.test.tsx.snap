// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportFileContent should render 1`] = `
<div>
  <div
    class="euiFilePicker euiFilePicker--large euiFilePicker--fullWidth"
  >
    <div
      class="euiFilePicker__wrap"
    >
      <input
        accept=".csv, .json, .ndjson"
        aria-describedby="mocked-id-filePicker__prompt"
        class="euiFilePicker__input"
        id="mocked-id"
        type="file"
      />
      <div
        class="euiFilePicker__prompt"
        id="mocked-id-filePicker__prompt"
      >
        <span
          aria-hidden="true"
          class="euiFilePicker__icon"
          data-euiicon-type="importAction"
        />
        <div
          class="euiFilePicker__promptText"
        >
          Select or drag and drop a .csv, .json, .ndjson file
        </div>
      </div>
    </div>
  </div>
</div>
`;
